src/cpx_genericcallbacks.jl

```diff
@@ -0,0 +1,260 @@
+                    callback_data.context, rcnt, nzcnt, rhs, string(sense), rmatbeg .-= 1, rmatind .-= 1, rmatval, purgeable, lcl)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+# Check is the callback for the context CPX_CALLBACKCONTEXT_CANDIDATE is invoked by a feasible integer point
+function cbcandidateispoint(callback_data::CallbackContext, is_point::Int)
+    return_status = @cpx_ccall(callbackcandidateispoint,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cint}),
+                    callback_data.context, is_point)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+# Check is the callback for the context CPX_CALLBACKCONTEXT_CANDIDATE is invoked by an unbounded relaxation
+function cbcandidateisray(callback_data::CallbackContext, is_ray::Int)
+    return_status = @cpx_ccall(callbackcandidateisray,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cint}),
+                    callback_data.context, is_ray)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetcandidatepoint(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int, obj::Cdouble)
+    return_status = @cpx_ccall(callbackgetcandidatepoint,
+                    Cint,
+                    (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint, Ref{Cdouble}),
+                    callback_data.context, x, begin_index - 1, end_index - 1, obj)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetcandidateray(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int)
+    return_status = @cpx_ccall(callbackgetcandidateray,
+                Cint,
+                (Ptr{Cvoid}, Ptr{Cdouble}, Cint, Cint),
+                callback_data.context, x, begin_index - 1, end_index - 1)
+    if return_status != 0
+        throw(CplexError(model.env, return_status))
+    end
+    return return_status
+end
+
+function cbgetfunc(model::Model, context_id::Clong, callback_func::Function)
+    c_callback = @cfunction(
+        callback_wrapper, Cint, (Ptr{Cvoid}, Clong, Ptr{Cvoid})
+    )
+    user_handle = (model, callback_func)::Tuple{Model, Function}
+    ret = @cpx_ccall(callbackgetfunc,
+        Cint,
+        (Ptr{Cvoid}, Ptr{Cvoid}, Ref{Clong}, Ptr{Cvoid}, Any),
+        model.env, model.lp, context_mask, c_callback, user_handle)
+    if ret != 0
+        throw(CplexError(model.env, ret))
+    end
+    model.callback = user_handle
+    return ret
+end
+
+function cbgetincumbent(callback_data::CallbackContext, x::Vector{Cdouble}, begin_index::Int, end_index::Int, obj::Cdouble)
+    return_status = @cpx_ccall(callbackgetincumbent,
+                    Cint,
```

Comment:
this indenting doesn't match the indenting in the other functions. I would recommend using this indenting (4 spaces only) in all the functions.